app/flatpak-builtins-install.c

```diff
@@ -395,6 +394,8 @@ flatpak_builtin_install (int argc, char **argv, GCancellable *cancellable, GErro
                   flatpak_split_partial_ref_arg_novalidate (argv[1], kinds, opt_arch, target_branch,
                                                             &matched_kinds, &id, &arch, &branch);
 
+                  g_clear_error (&local_error);
```

Comment:
I would much prefer a second local for this (say local_error2). The kind of non-locality we get by having a g_clear_error is both hard to read and easy to get wrong if code is moved around or inserted.

Or alternatively, we could move the first local error into `if (!auto_remote)`, which means its scope is more natural and we don't leave around a visible but possibly initialized local_error variable in the rest of the code.

Reply:
I tightened the scope on the error pointers, agree that's much better.